lib/experimental/cmdline/parsers.nim

```diff
@@ -0,0 +1,52 @@
+#
+#
+#                 NimSkull's runtime library
+#     (c) Copyright 2025 Leorize <leorize+oss@disroot.org>
+#
+# See the file "copying.txt", included in this distribution, for
+# details about copyright.
+
+## This module defines default command line parsers for primitive types
+
+import std/strutils
+import std/typetraits
+
+proc parseCli*(T: typedesc[SomeInteger], value: string): T =
```

Comment:
`tryParseCli` variants that don't raise but instead return a default value (taken as a parameter) would be handy, as people will be doing a lot of that.

Reply:
I'm not sure about the value of that for CLI purposes. More often than not you want to error out because wrong inputs from user meant that they make a mistake and shouldn't be ignored. For general parsing `parseutils` is what you want to use.

Other than that, a `try` variant never make much sense to me, as it'd require you to always construct a (potentially expensive) object regardless of parsing results. I could entertain the idea of using `Result[T]`, which would come with free support for defaults, but the ecosystem at the moment deals in exceptions.

Reply:
The idea with try variants is that you use in band error values to avoid the crappiness of `try...except` wrapping.

`Result[T]` is better, I keep forgetting we have one in the stdlib/experimental (I remember it as for the compiler). 😆

Reply:
I elected to not use `Result[T]` for now.

Most of the APIs an user would call to aid with parsing raises `ValueError` themselves, and at present this makes it a lot simpler to write a custom `parseCli` implementation, which I expect to happen more than explicitly invoking these.

We could add a capturing feature to `Result[T]` which would automatically turn exceptions into an error `Result`, which then an user can just do something like:

```nim
let val = capture(parseCli(int, "not int")).valueOr(-1)
```

and would work for all existing APIs, not just this one.